### ISSUE TITLE
no passed column in results

### ISSUE THREAD

**Issue by mradz19** [FrederickHuangLin/ANCOMBC#212]
I am running ancombc2 on the data outlined in the tutorial and my own data but I do not get any passed_ columns in my output object. I am following the tutorial exactly is the latest version still supposed to produce passed_ columns?

**Comment by RachBioHaz** [comment]
I'm also having issues since I upgraded R and RStudio.  Not sure if there is a compatibility issue.

**Comment by jorondo1** [comment]
same here! looking for the passed_ss column in a global test:

```
DA_global <- ancombc2(
  data = psMossMAGs, 
  tax_level= "Species",
  p_adj_method="holm", 
  prv_cut = 0.05, 
  fix_formula="Host + Compartment", 
  group = "Host", 
  struc_zero = TRUE,
  neg_lb = TRUE,
  pseudo_sens = TRUE,
  global = TRUE,
  pairwise = TRUE,
  alpha = 0.01,
  n_cl = 10,
  dunnet = TRUE, 
  verbose = TRUE
)

> DA_global$res_global %>% 
+                      dplyr::transmute(taxon, 
+                                       diff_Host = diff_abn,
+                                       passed_ss = passed_ss)
Error in `dplyr::transmute()`:
ℹ In argument: `passed_ss = passed_ss`.
Caused by error:
! object 'passed_ss' not found

```

**Comment by jorondo1** [comment]
> sessionInfo()
R version 4.2.2 (2022-10-31)
Platform: aarch64-apple-darwin20 (64-bit)
Running under: macOS Ventura 13.5.2

Matrix products: default
LAPACK: /Library/Frameworks/R.framework/Versions/4.2-arm64/Resources/lib/libRlapack.dylib

locale:
[1] en_US.UTF-8/en_US.UTF-8/en_US.UTF-8/C/en_US.UTF-8/en_US.UTF-8

attached base packages:
[1] parallel  stats4    stats     graphics  grDevices utils     datasets  methods  
[9] base     

other attached packages:
 [1] doRNG_1.8.6                 rngtools_1.5.2             
 [3] foreach_1.5.2               phyloseq_1.42.0            
 [5] bestNormalize_1.9.1         RColorBrewer_1.1-3         
 [7] vegan_2.6-4                 lattice_0.20-45            
 [9] permute_0.9-7               DESeq2_1.38.3              
[11] SummarizedExperiment_1.28.0 Biobase_2.58.0             
[13] MatrixGenerics_1.10.0       matrixStats_0.63.0         
[15] GenomicRanges_1.50.2        GenomeInfoDb_1.34.9        
[17] IRanges_2.32.0              S4Vectors_0.36.2           
[19] BiocGenerics_0.44.0         magrittr_2.0.3             
[21] lubridate_1.9.2             forcats_1.0.0              
[23] stringr_1.5.0               dplyr_1.1.1                
[25] purrr_1.0.1                 readr_2.1.4                
[27] tidyr_1.3.0                 tibble_3.2.1               
[29] ggplot2_3.4.1               tidyverse_2.0.0            
[31] ANCOMBC_2.0.3               pacman_0.5.1               

loaded via a namespace (and not attached):
  [1] utf8_1.2.3                     tidyselect_1.2.0              
  [3] lme4_1.1-32                    RSQLite_2.3.0                 
  [5] AnnotationDbi_1.60.2           htmlwidgets_1.6.2             
  [7] grid_4.2.2                     BiocParallel_1.32.6           
  [9] gmp_0.7-1                      munsell_0.5.0                 
 [11] ScaledMatrix_1.6.0             codetools_0.2-19              
 [13] future_1.32.0                  withr_2.5.0                   
 [15] colorspace_2.1-0               energy_1.7-11                 
 [17] knitr_1.42                     rstudioapi_0.14               
 [19] SingleCellExperiment_1.20.1    DescTools_0.99.48             
 [21] listenv_0.9.0                  labeling_0.4.2                
 [23] Rdpack_2.4                     emmeans_1.8.5                 
 [25] GenomeInfoDbData_1.2.9         farver_2.1.1                  
 [27] bit64_4.0.5                    rhdf5_2.42.0                  
 [29] coda_0.19-4                    parallelly_1.35.0             
 [31] vctrs_0.6.1                    treeio_1.22.0                 
 [33] generics_0.1.3                 ipred_0.9-14                  
 [35] xfun_0.38                      timechange_0.2.0              
 [37] R6_2.5.1                       doParallel_1.0.17             
 [39] ggbeeswarm_0.7.1               rsvd_1.0.5                    
 [41] locfit_1.5-9.7                 rhdf5filters_1.10.1           
 [43] bitops_1.0-7                   cachem_1.0.7                  
 [45] DelayedArray_0.24.0            vroom_1.6.1                   
 [47] scales_1.2.1                   nnet_7.3-18                   
 [49] beeswarm_0.4.0                 rootSolve_1.8.2.3             
 [51] gtable_0.3.3                   beachmat_2.14.0               
 [53] globals_0.16.2                 lmom_2.9                      
 [55] timeDate_4022.108              rlang_1.1.0                   
 [57] splines_4.2.2                  butcher_0.3.3                 
 [59] lazyeval_0.2.2                 checkmate_2.1.0               
 [61] reshape2_1.4.4                 backports_1.4.1               
 [63] Hmisc_5.0-1                    tools_4.2.2                   
 [65] lava_1.7.2.1                   decontam_1.18.0               
 [67] biomformat_1.26.0              proxy_0.4-27                  
 [69] plyr_1.8.8                     MultiAssayExperiment_1.24.0   
 [71] Rcpp_1.0.10                    base64enc_0.1-3               
 [73] sparseMatrixStats_1.10.0       zlibbioc_1.44.0               
 [75] RCurl_1.98-1.12                rpart_4.1.19                  
 [77] TreeSummarizedExperiment_2.6.0 viridis_0.6.2                 
 [79] ggrepel_0.9.3                  cluster_2.1.4                 
 [81] DECIPHER_2.26.0                data.table_1.14.8             
 [83] lmerTest_3.1-3                 mvtnorm_1.1-3                 
 [85] gsl_2.1-8                      hms_1.1.3                     
 [87] evaluate_0.20                  xtable_1.8-4                  
 [89] XML_3.99-0.14                  readxl_1.4.2                  
 [91] gridExtra_2.3                  compiler_4.2.2                
 [93] scater_1.26.1                  crayon_1.5.2                  
 [95] minqa_1.2.5                    htmltools_0.5.5               
 [97] tzdb_0.4.0                     mgcv_1.8-42                   
 [99] Formula_1.2-5                  geneplotter_1.76.0            
[101] expm_0.999-7                   Exact_3.2                     
[103] DBI_1.1.3                      MASS_7.3-58.3                 
[105] boot_1.3-28.1                  ade4_1.7-22                   
[107] Matrix_1.5-3                   cli_3.6.1                     
[109] rbibutils_2.2.13               igraph_1.4.1                  
[111] gower_1.0.1                    pkgconfig_2.0.3               
[113] numDeriv_2016.8-1.1            foreign_0.8-84                
[115] scuttle_1.8.4                  recipes_1.0.8                 
[117] annotate_1.76.0                vipor_0.4.5                   
[119] DirichletMultinomial_1.40.0    hardhat_1.3.0                 
[121] multtest_2.54.0                XVector_0.38.0                
[123] estimability_1.4.1             mia_1.6.0                     
[125] prodlim_2023.08.28             CVXR_1.0-11                   
[127] yulab.utils_0.0.6              digest_0.6.31                 
[129] Biostrings_2.66.0              rmarkdown_2.21                
[131] cellranger_1.1.0               tidytree_0.4.2                
[133] htmlTable_2.4.1                gld_2.6.6                     
[135] DelayedMatrixStats_1.20.0      nloptr_2.0.3                  
[137] lifecycle_1.0.3                nlme_3.1-162                  
[139] jsonlite_1.8.4                 Rhdf5lib_1.20.0               
[141] BiocNeighbors_1.16.0           viridisLite_0.4.1             
[143] fansi_1.0.4                    pillar_1.9.0                  
[145] KEGGREST_1.38.0                fastmap_1.1.1                 
[147] httr_1.4.5                     survival_3.5-5                
[149] glue_1.6.2                     png_0.1-8                     
[151] iterators_1.0.14               bit_4.0.5                     
[153] class_7.3-21                   stringi_1.7.12                
[155] blob_1.2.4                     BiocSingular_1.14.0           
[157] memoise_2.0.1                  Rmpfr_0.9-1                   
[159] irlba_2.3.5.1                  e1071_1.7-13                  
[161] future.apply_1.11.0            ape_5.7-1

**Comment by jorondo1** [comment]
Update: After upgrading to Ancom 2.4.0 this has been resolved (requires updating to R >= 4.3)

> sessionInfo()
R version 4.3.1 (2023-06-16)
Platform: aarch64-apple-darwin20 (64-bit)
Running under: macOS Ventura 13.5.2

Matrix products: default
BLAS:   /System/Library/Frameworks/Accelerate.framework/Versions/A/Frameworks/vecLib.framework/Versions/A/libBLAS.dylib 
LAPACK: /Library/Frameworks/R.framework/Versions/4.3-arm64/Resources/lib/libRlapack.dylib;  LAPACK version 3.11.0

locale:
[1] en_US.UTF-8/en_US.UTF-8/en_US.UTF-8/C/en_US.UTF-8/en_US.UTF-8

time zone: America/Montreal
tzcode source: internal

attached base packages:
[1] grid      stats4    stats     graphics  grDevices utils     datasets  methods   base     

other attached packages:
 [1] TreeSummarizedExperiment_2.10.0 SingleCellExperiment_1.24.0     cowplot_1.1.1                  
 [4] microViz_0.11.0                 trust_0.1-8                     data.table_1.14.8              
 [7] ComplexHeatmap_2.18.0           bestNormalize_1.9.1             RColorBrewer_1.1-3             
[10] vegan_2.6-4                     lattice_0.22-5                  permute_0.9-7                  
[13] DESeq2_1.42.0                   SummarizedExperiment_1.32.0     Biobase_2.62.0                 
[16] MatrixGenerics_1.14.0           matrixStats_1.0.0               GenomicRanges_1.54.1           
[19] GenomeInfoDb_1.38.0             IRanges_2.36.0                  S4Vectors_0.40.1               
[22] BiocGenerics_0.48.0             magrittr_2.0.3                  lubridate_1.9.3                
[25] forcats_1.0.0                   stringr_1.5.0                   dplyr_1.1.3                    
[28] purrr_1.0.2                     readr_2.1.4                     tidyr_1.3.0                    
[31] tibble_3.2.1                    ggplot2_3.4.4                   tidyverse_2.0.0                
[34] ANCOMBC_2.4.0                   phyloseq_1.46.0                 BiocManager_1.30.22            
[37] pacman_0.5.1                   

loaded via a namespace (and not attached):
  [1] fs_1.6.3                    bitops_1.0-7                DirichletMultinomial_1.44.0 httr_1.4.7                 
  [5] doParallel_1.0.17           numDeriv_2016.8-1.1         tools_4.3.1                 doRNG_1.8.6                
  [9] backports_1.4.1             utf8_1.2.4                  R6_2.5.1                    lazyeval_0.2.2             
 [13] mgcv_1.9-0                  GetoptLong_1.0.5            rhdf5filters_1.14.0         withr_2.5.2                
 [17] prettyunits_1.2.0           gridExtra_2.3               bayesm_3.1-6                cli_3.6.1                  
 [21] Cairo_1.6-1                 TSP_1.2-4                   sandwich_3.0-2              labeling_0.4.3             
 [25] robustbase_0.99-0           mvtnorm_1.2-3               proxy_0.4-27                yulab.utils_0.1.0          
 [29] foreign_0.8-85              scater_1.30.0               decontam_1.22.0             parallelly_1.36.0          
 [33] readxl_1.4.3                rstudioapi_0.15.0           RSQLite_2.3.2               shape_1.4.6                
 [37] generics_0.1.3              vroom_1.6.4                 gtools_3.9.4                Matrix_1.6-1.1             
 [41] biomformat_1.30.0           ggbeeswarm_0.7.2            fansi_1.0.5                 DescTools_0.99.50          
 [45] DECIPHER_2.30.0             abind_1.4-5                 lifecycle_1.0.3             multcomp_1.4-25            
 [49] Rtsne_0.16                  rhdf5_2.46.0                recipes_1.0.8               SparseArray_1.2.0          
 [53] blob_1.2.4                  crayon_1.5.2                butcher_0.3.3               beachmat_2.18.0            
 [57] pillar_1.9.0                knitr_1.45                  rjson_0.2.21                boot_1.3-28.1              
 [61] gld_2.6.6                   future.apply_1.11.0         codetools_0.2-19            glue_1.6.2                 
 [65] remotes_2.4.2.1             MultiAssayExperiment_1.28.0 png_0.1-8                   vctrs_0.6.4                
 [69] treeio_1.26.0               Rdpack_2.5                  cellranger_1.1.0            gtable_0.3.4               
 [73] cachem_1.0.8                gower_1.0.1                 xfun_0.40                   rbibutils_2.2.16           
 [77] S4Arrays_1.2.0              prodlim_2023.08.28          survival_3.5-7              timeDate_4022.108          
 [81] seriation_1.4.2             iterators_1.0.14            hardhat_1.3.0               lava_1.7.2.1               
 [85] bluster_1.12.0              gmp_0.7-2                   TH.data_1.1-2               ipred_0.9-14               
 [89] nlme_3.1-163                bit64_4.0.5                 rprojroot_2.0.3             tensorA_0.36.2             
 [93] irlba_2.3.5.1               vipor_0.4.5                 rpart_4.1.21                colorspace_2.1-0           
 [97] DBI_1.1.3                   Hmisc_5.1-1                 nnet_7.3-19                 ade4_1.7-22                
[101] processx_3.8.2              Exact_3.2                   tidyselect_1.2.0            curl_5.1.0                 
[105] bit_4.0.5                   compiler_4.3.1              compositions_2.0-6          microbiome_1.24.0          
[109] htmlTable_2.4.2             BiocNeighbors_1.20.0        expm_0.999-7                desc_1.4.2                 
[113] DelayedArray_0.28.0         checkmate_2.3.0             scales_1.2.1                DEoptimR_1.1-3             
[117] callr_3.7.3                 digest_0.6.33               minqa_1.2.6                 rmarkdown_2.25             
[121] ca_0.71.1                   XVector_0.42.0              htmltools_0.5.6.1           pkgconfig_2.0.3            
[125] base64enc_0.1-3             lme4_1.1-34                 sparseMatrixStats_1.14.0    fastmap_1.1.1              
[129] GlobalOptions_0.1.2         rlang_1.1.1                 htmlwidgets_1.6.2           DelayedMatrixStats_1.24.0  
[133] farver_2.1.1                zoo_1.8-12                  jsonlite_1.8.7              energy_1.7-11              
[137] BiocParallel_1.36.0         BiocSingular_1.18.0         RCurl_1.98-1.12             Formula_1.2-5              
[141] scuttle_1.12.0              GenomeInfoDbData_1.2.11     Rhdf5lib_1.24.0             munsell_0.5.0              
[145] Rcpp_1.0.11                 ape_5.7-1                   viridis_0.6.4               CVXR_1.0-11                
[149] stringi_1.7.12              rootSolve_1.8.2.4           zlibbioc_1.48.0             MASS_7.3-60                
[153] pkgbuild_1.4.2              plyr_1.8.9                  parallel_4.3.1              listenv_0.9.0              
[157] ggrepel_0.9.4               lmom_3.0                    Biostrings_2.70.1           splines_4.3.1              
[161] multtest_2.58.0             circlize_0.4.15             hms_1.1.3                   locfit_1.5-9.8             
[165] ps_1.7.5                    igraph_1.5.1                rngtools_1.5.2              reshape2_1.4.4             
[169] ScaledMatrix_1.10.0         evaluate_0.22               nloptr_2.0.3                tzdb_0.4.0                 
[173] foreach_1.5.2               clue_0.3-65                 future_1.33.0               rsvd_1.0.5                 
[177] Rmpfr_0.9-3                 e1071_1.7-13                tidytree_0.4.5              viridisLite_0.4.2          
[181] class_7.3-22                gsl_2.1-8                   lmerTest_3.1-3              registry_0.5-1             
[185] memoise_2.0.1               beeswarm_0.4.0              cluster_2.1.4               timechange_0.2.0           
[189] globals_0.16.2              mia_1.10.0

**Comment by FrederickHuangLin** [comment]
Hi @mradz19, @RachBioHaz, and @jorondo1! 

Thanks for your inquiry! Yes, as reported by @jorondo1, you will have to upgrade the `ANCOMBC` version to access the desired columns.

Best,
Frederick
